lib/application.js

```diff
@@ -372,7 +372,14 @@ app.enableAuth = function(options) {
               ctx.args.filter = { fields: {} };
             }
             if (typeof ctx.args.filter === 'string') {
-              ctx.args.filter = JSON.parse(ctx.args.filter);
+              try {
+                ctx.args.filter = JSON.parse(ctx.args.filter);
+              } catch (error) {
+                var e = new SyntaxError('Bad request: Invalid JSON.');
```

Comment:
Could you just write it as https://github.com/weflex/neaty/blob/fix/catch-json-parse/lib/application.js#L396-L413?